src/tools/fuzzing/fuzzing.cpp

```diff
@@ -2178,15 +2178,18 @@ Expression* TranslateToFuzzReader::makeConstCompoundRef(Type type) {
   // really have no choice and must emit a null (or else we could infinitely
   // recurse). For the nesting limit, use a bound that is higher than the normal
   // one, so that the normal mechanisms should prevent us from getting here;
-  // this limit is really a last resort we want to never reach.
+  // this limit is really a last resort we want to never reach. Also, increase
+  // the chance to emit a null as |nesting| rises, to avoid deep recursive
+  // structures.
   //
   // Note that we might have cycles of types where some are non-nullable. We
   // will only stop here when we exceed the nesting and reach a nullable one.
   // (This assumes there is a nullable one, that is, that the types are
   // inhabitable.)
-  const auto LIMIT = 2 * NESTING_LIMIT;
+  const auto LIMIT = NESTING_LIMIT + 1;
```

Comment:
Why the `+ 1`? It looks like just reusing `NESTING_LIMIT` would work as well.

Reply:
As the comment above says, this tries to be higher than that limit so that the normal nesting-limiting mechanisms can kick in first, hopefully. They are less "severe", in that here we'll emit a null which is likely to cause a trap if it is used.